File: db/migrate/0006_create_media.rb
# -*- coding: utf-8 -*-

class CreateMedia < ActiveRecord::Migration
  def change
    create_table(:media) do |t|
      t.string :name, null: false
      t.string :media_type, null: false
      t.string :url, null: false
      t.string :path, null: false
    end

    add_index :media, :url
    add_reference :media, :blog, null: false
    add_index :media, [:blog_id, :url], unique: true
  end
end

# eof

File: priv/repo/migrations/0006_create_media.exs
defmodule Burp.Repo.Migrations.CreateMedia do
  use Ecto.Migration

  def change do
    create table(:media) do
      add :name, :string, null: false
      add :media_type, :string, null: false
      add :url, :string, null: false
      add :path, :string, null: false
      add :blog_id, references(:blogs, on_delete: :nothing), null: false

      timestamps()
    end

    create index(:media, [:blog_id])
    create index(:media, [:url])
    create unique_index(:media, [:blog_id, :url])
  end
end
